core/src/main/java/org/apache/carbondata/core/scan/expression/RangeExpressionEvaluator.java

```diff
@@ -223,12 +223,24 @@ private void fillExpressionMap(Map<String, List<FilterModificationNode>> filterE
       addFilterExpressionMap(filterExpressionMap, currentNode, parentNode);
     }
 
-    for (Expression exp : currentNode.getChildren()) {
-      if (null != exp) {
-        fillExpressionMap(filterExpressionMap, exp, currentNode);
-        if (exp instanceof OrExpression) {
-          replaceWithRangeExpression(filterExpressionMap);
-          filterExpressionMap.clear();
+    // In case of Or Exp we have to evaluate both the subtrees of expression separately
+    // else it will combine the results of both the subtrees into one expression
+    // which wont give us correct result
+    if (currentNode instanceof OrExpression) {
+      if (null != ((OrExpression) currentNode).left) {
+        fillExpressionMap(filterExpressionMap, ((OrExpression) currentNode).left, currentNode);
```

Comment:
maybe there were some nodes in filterExpressionMap already.

Reply:
so it still faces the same issue like before

Reply:
I have created a separate map in case of Or expression's evaluation. Now it shall work fine and cover all the cases.